core_api/src/lib.rs

```diff
@@ -194,15 +224,49 @@ mod tests {
                 assert_eq!(hc.instance.state().nucleus().dna(), Some(dna));
                 assert!(!hc.active);
                 assert_eq!(hc.context.agent, agent);
-                // TODO #61 - Should instantiation also initialize?
-                // assert!(hc.instance.state().nucleus().has_initialized());
+                assert!(hc.instance.state().nucleus().has_initialized());
                 let test_logger = test_logger.lock().unwrap();
                 assert_eq!(format!("{:?}", *test_logger), "\"TestApp instantiated\"");
             }
             Err(_) => assert!(false),
         };
     }
 
+    #[test]
+    fn fails_instantiate_if_genesis_fails() {
+        let mut dna = create_test_dna_with_wat(
+            "test_zome".to_string(),
+            ReservedCapabilityNames::LifeCycle.as_str().to_string(),
+            Some(
+                r#"
+            (module
+                (memory (;0;) 17)
+                (func (export "genesis_dispatch") (param $p0 i32) (param $p1 i32) (result i32)
+                    i32.const 4
+                )
+                (data (i32.const 0)
+                    "fail"
+                )
+                (export "memory" (memory 0))
+            )
+        "#,
+            ),
+        );
+
+        dna.name = "TestApp".to_string();
+        let agent = HCAgent::from_string("bob");
+        let (context, _test_logger) = test_context(agent.clone());
+        let result = Holochain::new(dna.clone(), context.clone());
+
+        match result {
+            Ok(_) => assert!(false),
+            Err(err) => assert_eq!(err, HolochainError::ErrorGeneric("fail".to_string())),
+        };
+    }
+
+    #[test]
+    fn fails_instantiate_if_genesis_times_out() {}
```

Comment:
Do want to implement this test?

Either implement or remove, please.

Reply:
oh yah, will do.

Reply:
done